app/src/main/java/xyz/jhughes/laundry/adapters/MachineAdapter.java

```diff
@@ -135,24 +145,134 @@ public void onClick(DialogInterface dialog, int which) {
+        final Handler handler = new Handler();
+        lockListener.onLock();
+        AlertDialog.Builder machineWaitingDialog = new AlertDialog.Builder(mContext);
+        machineWaitingDialog.setTitle(mContext.getString(R.string.alarm))
+                .setMessage(mContext.getString(R.string.available_timer_message1) + " " + m.getName() + " " + mContext.getString(R.string.available_timer_message2))
+                .setCancelable(true)
+                .setPositiveButton(mContext.getString(R.string.available_timer_refresh), null)
+                .setNegativeButton(mContext.getString(R.string.available_timer_cancel), new DialogInterface.OnClickListener() {
+                    @Override
+                    public void onClick(DialogInterface dialog, int id) {
+                        handler.removeCallbacksAndMessages(null);
+                        dialog.cancel();
+                    }
+                });
+        //add message to load machine before
+        LayoutInflater inflater = LayoutInflater.from(mContext);
+        View q = inflater.inflate(R.layout.view_available_machine, null);
+        machineWaitingDialog.setView(q);
+        TextView number = (TextView) q.findViewById(R.id.machine_name_number);
+        number.setText(getNumberFromName(m));
+        final AlertDialog alertDialog = machineWaitingDialog.create();
+        alertDialog.show();
+        alertDialog.setOnCancelListener(new DialogInterface.OnCancelListener() {
+            @Override
+            public void onCancel(DialogInterface dialog) {
+                lockListener.onUnlock();
+            }
+        });
+        alertDialog.getButton(AlertDialog.BUTTON_POSITIVE).setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                String apiLocationFormat = Constants.getApiLocation(MachineAdapter.this.roomName);
+                Call<ArrayList<Machine>> call = MachineService.getService().getMachineStatus(apiLocationFormat);
+                call.enqueue(new Callback<ArrayList<Machine>>() {
+                    @Override
+                    public void onResponse(Call<ArrayList<Machine>> call, Response<ArrayList<Machine>> response) {
+                        ArrayList<Machine> body = response.body();
+                        if (body.contains(m2)){
+                            Machine m3 = body.get(body.indexOf(m2));
+                            if (m3.getStatus().equals(MachineStates.IN_USE)){ //
+                                createNotification(m3);
+                                alertDialog.cancel();
+                            }
+                        }
+                    }
+
+                    @Override
+                    public void onFailure(Call<ArrayList<Machine>> call, Throwable t) {
+                    }
+                });
+            }
+        });
+
+        handler.postDelayed(new MachineCheckerRunnable(m, this.roomName, handler, new OnMachineChangedToInUse() {
```

Comment:
Can we consider making this handler into a class of its own called something like "MachineStatusChecker". Then we arent interacting with the handler directly here. We can just have an instance of "MachineStatucChecker" that has "startChecking()" and "stopChecking()" methods on it. 

Rather than all of these handler specific method calls.